photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/NameAndTerm.scala

```diff
@@ -18,13 +18,21 @@ package com.linkedin.photon.ml.avro.data
  * A compact way to represent the feature key as (name, term) pair.
  * @author xazhang
  */
-//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]] after Avro related classes/functons are decoupled from the rest of code
+//TODO: Change the scope of this class and all functions in the companion object to [[com.linkedin.photon.ml.avro]]
+//after Avro related classes/functons are decoupled from the rest of code
 protected[ml] case class NameAndTerm(name: String, term: String) {
 
   override def hashCode: Int = {
     (name + NameAndTerm.DELIMITER + term).hashCode
   }
 
+  override def equals(that: Any): Boolean = {
+    that match {
+      case other: NameAndTerm => hashCode == other.hashCode
```

Comment:
When I looked at this change, the first two things came to my mind were Long and String. In some cases, two longs (or two strings) may share the same hash code, but they are still different when compared by the ```equals``` method, because the ```equals``` method has nothing to do with the ```hashCode``` in those two cases.

For NameAndTerm, it appears to me that two NameAndTerm instances may have hash code collision (due to String hash code collision) in some rare situations, but they still should be considered as different NameAndTerm instances.

Anyway, I feel like I am now trying to dig into some small details that may not matter at all in real life. Please go ahead and merge this RB when it's ready. Thanks for fixing all the Scala stye issues by the way :)